Document using Cypher 5 for NOM persistence (#262)
neo4j/docs-ops-manager: modules/ROOT/pages/installation/persistence.adoc

The first step, as described in this section, is to set up a Neo4j DBMS to act as persistence storage for NOM.

> The NOM persistence DBMS is used to store all NOM data (topology, metrics, etc.)

Reviewer: ```suggestion
The NOM persistence DBMS is used to store all NOM data (topology, metrics, etc.).
```

The DBMS and the NOM Server need to be running simultaneously, or the NOM Server will not work.

== Self-managed DBMS

NOM comes with a license to run a Neo4j DBMS with up to 4 cores of CPU and up to 16 GB of RAM for the use as NOMs persistence.

=== Specifications

> * Neo4j Enterprise Edition 4.4, 5.x or 2025.x

Reviewer: ```suggestion
* Neo4j Enterprise Edition 4.4, 5.x, or 2025.x
```

* At least 10 GB free disk space
* For minimum requirements, see also link:/docs/operations-manual/current/installation/requirements/[Operations Manual -> System requirements].

=== Install and prepare

> Please refer to the link:/docs/operations-manual/current/installation[Operations Manual -> Installation] for information on how to install Neo4j.

Reviewer: ```suggestion
Refer to the link:{neo4j-docs-base-uri}/operations-manual/current/installation[Operations Manual -> Installation] for information on how to install Neo4j.
```

Ensure that the Neo4j database to be used for NOM persistence fulfills the following requirements:

> * it is empty and does not contain any indexes or constraints,

Reviewer: ```suggestion
* The database is empty and does not contain any indexes or constraints.
```

> * Cypher 5 is selected as default language for this database. For example, this can be done by executing the statement `ALTER DATABASE neo4j SET DEFAULT LANGUAGE CYPHER 5`. See more details in link:/docs/cypher-manual/current/queries/select-version/#select-default-cypher-version[Cypher Manual -> Select Cypher version].

Reviewer: ```suggestion
* Cypher 5 is selected as default language for this database. 
For example, this can be done by executing the statement `ALTER DATABASE neo4j SET DEFAULT LANGUAGE CYPHER 5`. 
For more details see link:{neo4j-docs-base-uri}/cypher-manual/current/queries/select-version/#select-default-cypher-version[Cypher Manual -> Select Cypher version].
```

> NOM server installation requires specifying credentials of a Neo4j user. Ensure that it fulfills the following requirements:

Reviewer: ```suggestion
NOM server installation requires specifying credentials of a Neo4j user. 
Ensure that it fulfills the following requirements:
```

> * if a build-in admin user (`neo4j`) is used, ensure it has a secure password,

Reviewer: ```suggestion
* If a built-in admin user (`neo4j`) is used, ensure it has a secure password.
```

> * if a new user is created, ensure it has full admin access.

Reviewer: ```suggestion
* If a new user is created, ensure it has full admin access.
```

The DBMS and the NOM Server need to be running simultaneously, or the NOM Server will not work.

== Self-managed DBMS

NOM comes with a license to run a Neo4j DBMS with up to 4 cores of CPU and up to 16 GB of RAM for the use as NOMs persistence.

=== Specifications
* Neo4j Enterprise Edition 4.4, 5.x, or 2025.x
* At least 10 GB free disk space
* For minimum requirements, see also link:/docs/operations-manual/current/installation/requirements/[Operations Manual -> System requirements].

=== Install and prepare

> Refer to the link:{neo4j-docs-base-uri}/operations-manual/current/installation[Operations Manual -> Installation] for information on how to install Neo4j.

Reviewer: ```suggestion
Refer to the link:https://neo4j.com/docs/operations-manual/current/installation/)[Operations Manual -> Installation] for information on how to install Neo4j.
```

Ensure that the Neo4j database to be used for NOM persistence fulfills the following requirements:

* The database is empty and does not contain any indexes or constraints.
* Cypher 5 is selected as default language for this database. 
For example, this can be done by executing the statement `ALTER DATABASE neo4j SET DEFAULT LANGUAGE CYPHER 5`. 

> For more details see link:{neo4j-docs-base-uri}/cypher-manual/current/queries/select-version/#select-default-cypher-version[Cypher Manual -> Select Cypher version].

Reviewer: ```suggestion
For more details see link:https://neo4j.com/docs/cypher-manual/current/queries/select-version/[Cypher Manual -> Select Cypher version].
```
